qutip/bloch.py

```diff
@@ -38,16 +38,28 @@
 from numpy import (ndarray, array, linspace, pi, outer, cos, sin, ones, size,
                    sqrt, real, mod, append, ceil, arange)
 
+from packaging.version import parse as parse_version
+
 from qutip.qobj import Qobj
 from qutip.expect import expect
 from qutip.operators import sigmax, sigmay, sigmaz
 
 try:
+    import matplotlib
     import matplotlib.pyplot as plt
     from mpl_toolkits.mplot3d import Axes3D
     from matplotlib.patches import FancyArrowPatch
     from mpl_toolkits.mplot3d import proj3d
 
+    if parse_version(matplotlib.__version__) >= parse_version('3.4'):
+        def _axes3D(*args, **kwargs):
+            fig = args[0]
```

Comment:
It's better to spell this as
```python
def _axes3D(fig, *args, **kwargs):
    ...
```
because then if there's an error calling the function (i.e. by passing _no_ arguments), you get the "normal" Python error
```python
TypeError: _axes3D() missing 1 required positional argument: 'fig'
```
What I wrote is a pretty common Python idiom when you want to pick out the first positional argument (or more).  You can similarly pick out special keyword arguments with `def xxx(*args, out=None, linewidth=None, **kwargs)`.

It also would be good if you could write a little comment above the `if parse_version()` line (just a one- or two-line comment) explaining why this is necessary; we understand right now why it was put in place, but the next people to look at the code won't have the context of this PR.

Reply:
(and a similar comment at the visualisation.py change as well)

Reply:
Thanks @jakelishman for all these suggestions! I have made the necessary changes.